lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -512,8 +514,9 @@ def _on_opensearch_data_storage_detaching(self, _: StorageDetachingEvent):  # no
                 else:
                     raise OpenSearchHAError(ClusterHealthUnknown)
         finally:
-            # release lock
-            self.node_lock.release()
+            if self.app.planned_units() > 0 and (self.opensearch.is_node_up() or self.alt_hosts):
```

Comment:
Same comment as above.

Reply:
Similarly to the comments above. I am just not 100% sure if we should have it as `> 0` or  `> 1`.